doc/memos/rdm-draft-alamos-ieee802154-radio-hal.md

```diff
@@ -0,0 +1,908 @@
+requires that the transceiver is in `TX_ON` state before transmitting the
+packet and will not go back to `RX_ON` state unless the upper layer explicitly
+requests it. This is desired because the 802.15.4 MAC layer requires full
+control of the transceiver states.
+
+Also, this approach presents some advantages:
+- It makes network stack integration easier if the network stack requires direct
+  access to the radio (OpenWSN)
+- It avoids unnecesary state changes. It is possible to set the `TX_ON` state
+  once and sequentially send packets. When the upper layer finishes sending
+  the batch of packets, it can manually request the radio to set the `RX_ON`
+  state.
+
+## 5.3 Prepare and Transmit
+The Radio HAL doesn't define an explicit send function. Unlike the `netdev`
+approach, it bases on separation of the send procedure into frame loading
+and triggering the transmissions start.
+
+Although it is possible to load and start using the `netdev` interface with the
+`NETOPT_PRELOADING` option, the Radio HAL approach is easier and more
+lightweight to implement since it doesn't require internal state variables.
+
+Separated load and start is required for MAC layers
+with timing constraints (E.g. TSCH mode of 802.15.4 MAC). 
+
+In the rare case a radio device doesn't support loading the frame buffer without
+triggering a transmission, it is still possible to implement the load and
+transmit pattern using an internal buffer. However, this case is very unlikely
+because such a device could not meet 802.15.4 timing requirements.
+
+It is expected that a higher layer "send" function is defined for convenience
+which handles both loading and frame sending. Typically, this would be a
+802.15.4 MAC implementation which preloads the devices buffer once accessible,
+and triggers a send operation at a scheduled time slot. Alternatively, this
+could be a helper function for non MAC users.
+
+## 5.4 TX and RX Information
+Sometimes upper layers require information associated to the transmission
+or reception of a packet. The TSCH mode of the 802.15.4 MAC may require
+LQI and RSSI data from a received packet to schedule new cells.
+The 802.15.4 MAC may also require the information associated to the
+frame retransmission component (frame pending bit, number of retransmission,
+status) if the hardware provides support for hardware retransmissions.
+
+The 802.15.4 Radio HAL API provides functions to retrieve this data.
+Note that retrieving this information is optional in cases where the RX
+information is not needed or when the device doesn't support frame
+retransmissions.
+
+## 5.5 Thread Safety
+
+The 802.15.4 Radio HAL is not thread safe. As a consecuence of this, it is
+required that the Bottom-Half processor and all function calls to the Radio HAL
+API run in the same thread context. This is the same approach of the `netdev`
+interface.
+
+As a future work this work can be adapted to be thread safe.
```

Comment:
Sure this is worth the effort? Anyway, :+1: for not worrying about a thread safe interface now.